src/active/clojure/openid.clj

```diff
@@ -301,21 +301,23 @@
                          time/from-now)
                      (dissoc body :access-token :token-type :expires-in :refresh-token :id-token)))
 
-(defn fetch-access-token!
+(defn- fetch-access-token!
   "For a `openid-profile` and based on a `request` (the response of the
   idp), fetch the actual (JWT) access token.
 
   Might throw an exception."
-  [openid-profile authorization-code & [redirect-uri]]
+  [openid-profile authorization-code redirect-uri grant-type scope]
   (let [access-token-uri (lens/yank openid-profile (lens/>> openid-profile-openid-provider-config
                                                             openid-provider-config-token-endpoint))
         client-id        (openid-profile-client-id openid-profile)
         client-secret    (openid-profile-client-secret openid-profile)
-        payload          {:form-params {:grant_type   "authorization_code"
-                                        :code         authorization-code
-                                        :redirect_uri (absolute-redirect-uri openid-profile redirect-uri)
-                                        :client_id     client-id
-                                        :client_secret client-secret}}
+        payload          {:form-params (merge
+                                        {:client_id     client-id
+                                         :client_secret client-secret
+                                         :grant_type (or grant-type "authorization_code")}
```

Comment:
Since `grant_type` now is a mandatory argument (as it correctly should be), the `(or ... "authorization_code")` is no longer needed.  Please remove that.

Reply:
Good catch, fixed it.